examples/algorithms/plot_dedensification.py

```diff
@@ -0,0 +1,103 @@
+"""
+=============
+Dedensification
+=============
+
+Examples of dedensification of a graph.  Dedensification retains the structural
+pattern of the original graph and will only add compressor nodes when doing so
+would result in fewer edges in the compressed graph.
+"""
+
+import networkx as nx
+from networkx.algorithms import summarization
+import matplotlib.pyplot as plt
+
+plt.suptitle("Dedensification")
+
+original_graph = nx.DiGraph()
+white_nodes = ["1", "2", "3", "4", "5", "6"]
+red_nodes = ["A", "B", "C"]
+node_sizes = [250 for node in white_nodes + red_nodes]
+node_colors = []
+for node in white_nodes:
+    node_colors.append("white")
+for node in red_nodes:
+    node_colors.append("red")
+
+original_graph.add_nodes_from(white_nodes + red_nodes)
+original_graph.add_edges_from(
+    [
+        ("1", "C"),
+        ("1", "B"),
+        ("2", "C"),
+        ("2", "B"),
+        ("2", "A"),
+        ("3", "B"),
+        ("3", "A"),
+        ("3", "6"),
+        ("4", "C"),
+        ("4", "B"),
+        ("4", "A"),
+        ("5", "B"),
+        ("5", "A"),
+        ("6", "5"),
+        ("A", "6"),
+    ]
+)
+base_options = dict(with_labels=True, edgecolors="black")
+pos = {
+    "3": (0, 1),
+    "2": (
+        0,
+        2,
+    ),
+    "1": (
+        0,
+        3,
+    ),
```

Comment:
I'm not sure why these lines are formatted this way -- seems like they could be one line per node like the others.
If `black` did this I think you could put them on one line and black would not re-split them.  Worth a try anyway.